molecule_vagrant/modules/vagrant.py

```diff
@@ -508,79 +351,57 @@ def _get_config(self):
         return molecule.config.Config(molecule_file)
 
     def _write_vagrantfile(self):
+        # Helper lambdas to get the CPU and memory fields for each provider
+        cpu_field = (
+            lambda provider: "vmx['numvcpus']"
+            if provider.startswith("vmware_")
+            else "cpus"
+        )
+        memory_field = (
+            lambda provider: "vmx['memsize']"
+            if provider.startswith("vmware_")
+            else "memory"
+        )
+
         template = molecule.util.render_template(
             VAGRANTFILE_TEMPLATE,
-            vagrantfile_config=self._config.driver.vagrantfile_config,
+            cpu_field=cpu_field,
+            memory_field=memory_field,
+            provider=self._module.params["provider"],
+            instances=self._get_vagrant_instances(),
         )
         molecule.util.write_file(self._vagrantfile, template)
 
-    def _write_vagrantfile_config(self, data):
-        molecule.util.write_file(
-            self._config.driver.vagrantfile_config, molecule.util.safe_dump(data)
-        )
-
-    def _write_configs(self):
-        self._write_vagrantfile_config(self._get_vagrant_config_dict())
-        self._write_vagrantfile()
-
     def _get_vagrant(self):
-        v = vagrant.Vagrant(
+        return vagrant.Vagrant(
```

Comment:
Shouldn't this kind of change go to the cleanup patch ?

Reply:
Fixed in 69ba9bd.